src/java/org/apache/cassandra/db/compaction/CompactionCursor.java

```diff
@@ -439,12 +416,44 @@ else if (UnfilteredSerializer.isTombstoneMarker(flags)) {
         return partitionWritten;
     }
 
+    private DeletionTime maybePurgedOutputDeletion(DeletionTime mergedDeletion) throws IOException
+    {
+        final DeletionTime toWritePartitionDeletion;
+
+        if (!mergedDeletion.isLive() && !purger.shouldPurge(mergedDeletion))
+        {
+            toWritePartitionDeletion = mergedDeletion;
+            maybeSwitchWriter(compactionAwareWriter);
```

Comment:
I know this may be slightly less efficient, but I would prefer the writing to be done by the caller, rechecking `isLive` on the purged deletion, for clarity.

Alternatively, rename the method to something that clarifies that it may also write the partition header.